### PR TITLE
Moving Wrapping Outside of Vision Explanation Methods

### PR DESCRIPTION
**Pull request by natalie-isak**
This PR removes the custom ml wrapper class to the generic ml-wrapper repo. This has two implications:
1. ensures configurability and simplicity for future users of the wrapping class. 
2. removes the wrapping step each time the drise wrapper is called

### CODE REVIEW

File: requirements-test.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 pytest
-pytest-cov
\ No newline at end of file
+pytest-cov
+ml_wrappers
```
Comment (Advitya17) [inline requirements-test.txt:3]: is the name of the pip package `ml_wrappers` or `ml-wrappers`? Not sure if I've ever seen a python module with `_` in its names being pip installed.

File: tests/test_vision_explanation.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 import urllib.request as request_file
 
 import matplotlib.pyplot as plt
+import torch
```
Comment (Advitya17) [inline tests/test_vision_explanation.py:11]: Consider trapping import error or perhaps updating README that torch is not installed from dependencies file and should be installed from pytorch's official website - https://pytorch.org/get-started/locally/

File: tests/test_vision_explanation.py [hunk 3]
```diff
@@ -37,7 +38,7 @@ def test_vision_explain_preloaded():
     # save tested result in res
 
     # run the main function for saliency map generation
-    res = dr.get_drise_saliency_map(imgpath, None, None, None, savepath)
+    res = dr.get_drise_saliency_map(imgpath, None, None, savepath)
```
Comment (Advitya17) [inline tests/test_vision_explanation.py:41]: Nit: may be good to have arg names here to describe the different params set to None.

**Review by Advitya17 — APPROVED** [review]

File: requirements-test.txt (hunk 1 above)
Comment (natalie-isak) [reply requirements-test.txt:3]: It is ml_wrappers! Haha I've actually never seen python modules with "-"

File: tests/test_vision_explanation.py (hunk 2 above)
Comment (natalie-isak) [reply tests/test_vision_explanation.py]: trapped torch and vision explanation methods

File: requirements-test.txt [hunk 4]
```diff
@@ -1,2 +1,4 @@
...
```
Comment (imatiach-msft) [inline requirements-test.txt:3]: these should be in package dependencies and not test

File: requirements-test.txt [hunk 5]
```diff
@@ -1,2 +1,4 @@
...
+vision_explanation_methods
```
Comment (imatiach-msft) [inline requirements-test.txt:4]: this is the vision-explanation-methods package - it is already installed from: 
```
pip install -e .
```
hence, this line can just be removed

File: python/vision_explanation_methods/DRISE_runner.py [hunk 6]
```diff
@@ -208,39 +108,18 @@ def get_drise_saliency_map(
...
-            print("loading user fastercnn model")
-            model = get_instance_segmentation_model(numclasses)
-            model.load_state_dict(
-                torch.load(modellocation, map_location=device))
-    else:
-        if modellocation:
-            print("loading any user model")
-            model.load_state_dict(
-                torch.load(modellocation, map_location=device))
+        print("using pretrained fastercnn model")
```
Comment (imatiach-msft) [inline python/vision_explanation_methods/DRISE_runner.py:111]: remove print

File: tests/test_vision_explanation.py [hunk 7]
```diff
@@ -100,7 +118,8 @@ def test_vision_explain_loadmodel():
     _ = download_assets(modelpath)
 
     # run the main function for saliency map generation
-    res = dr.get_drise_saliency_map(imgpath, None, modelpath, 5, savepath)
+    model = (torch.load(modelpath, map_location='cpu'))
```
Comment (Advitya17) [inline tests/test_vision_explanation.py:121]: Should we replace map location with `cuda if torch.cuda.is_available() else cpu` ?

File: tests/test_vision_explanation.py [hunk 8]
```diff
@@ -100,7 +118,8 @@ def test_vision_explain_loadmodel():
...
+    res = dr.get_drise_saliency_map(imgpath, model, 5, savepath)
```
Comment (Advitya17) [inline tests/test_vision_explanation.py:122]: These 4 args look like the `modellocation` is 5 instead of `numclasses` (ref - lines 147-151), which may be incorrect. Suggestion having attribute names for each of the args.

**Review by Advitya17 — APPROVED** [review]

File: requirements-test.txt (hunk 4 above)
Comment (natalie-isak) [reply requirements-test.txt:3]: added to requirements.txt in the outer level of this repo and removed from test.

File: python/vision_explanation_methods/version.py [hunk 9]
```diff
@@ -3,5 +3,5 @@
 name = 'vision_explanation_methods'
 _major = '0'
 _minor = '0'
-_patch = '3'
+_patch = '4'
```
Comment (imatiach-msft) [inline python/vision_explanation_methods/version.py:6]: can you move this version update to a separate PR?

File: python/vision_explanation_methods/explanations/common.py [hunk 10]
```diff
@@ -247,11 +247,11 @@ def expand_class_scores(
     """
     number_of_detections = scores.shape[0]
 
-    expanded_scores = torch.ones(number_of_detections, number_of_classes)
+    expanded_scores = torch.ones(number_of_detections, number_of_classes + 1)
```
Comment (imatiach-msft) [inline python/vision_explanation_methods/explanations/common.py:250]: why did this file need to be changed?  it looks like expanded_scores and residual were changed.

File: python/vision_explanation_methods/DRISE_runner.py [hunk 11]
```diff
@@ -171,14 +76,12 @@ def get_instance_segmentation_model(num_classes: int):
...
         imagelocation: str,
         model: Optional[object],
-        modellocation: Optional[str],
         numclasses: int,
         savename: str,
         nummasks: int = 25,
         maskres: Tuple[int, int] = (4, 4),
         maskpadding: Optional[int] = None,
-        devicechoice: Optional[str] = None,
-        wrapperchoice: Optional[object] = PytorchFasterRCNNWrapper
```
Comment (tongyu-microsoft) [inline python/vision_explanation_methods/DRISE_runner.py:181]: remove line 105-106?

File: tests/test_vision_explanation.py [hunk 12]
```diff
@@ -100,7 +129,24 @@ def test_vision_explain_loadmodel():
     _ = download_assets(modelpath)
 
     # run the main function for saliency map generation
-    res = dr.get_drise_saliency_map(imgpath, None, modelpath, 5, savepath)
+    # empty_model = PytorchDRiseWrapper()
```
Comment (tongyu-microsoft) [inline tests/test_vision_explanation.py:132]: remove these commented code?

**Review by tongyu-microsoft — APPROVED** [review]
